wire-ios/Wire-iOS/Sources/UserInterface/DeviceView/Views/E2EIdentityCertificateDetailsView.swift

```diff
@@ -51,7 +51,8 @@ struct E2EIdentityCertificateDetailsView: View {
                         didDismiss?()
                     },
                     label: {
-                        Image(.close)
+                        Image
+                            .close
```

Comment:
Is this minor difference really worth spoiling the `Image` namespace?
`Image(.close)` vs. `Image.close`
First, I'd consider the first variant more readable, everybody is familiar with the `init` taking an `ImageResource`.
Second, it doesn't scale well. Imagine having all images in the codebase as extension to `SwiftUI.Image`. A long list of assets mixed with the original API.

In my opinion the `ImageResource` is a great shortcut on its own.

Reply:
I see, what about creating a `CloseButton` view in `ReuisableUIComponents`?

Reply:
The `CloseButton` itself already exists so I could probably update `E2EIdentityCertificateDetailsView.swift` to use it. So then the pattern would be for the close icon to be in the `ReuisableUIComponents` target, not in `WireDesign`. 

I think that should work in this case but I'm worried it is too rigid. It basically means we should never add image assets to `WireDesign` right?

Reply:
Interestingly a public `ImageResource` has been added to iOS 17 but it still only generates internal assets. It does mean that something like this is possible from iOS 17. Still not great:

```swift
enum Icon {
    static let close = ImageResource.close
}

let image = Image(Icon.close)
```

Reply:
@caldrian I'll look into improving this in a separate PR. Maybe we can discuss the topic a bit further on Friday... Essentially where should image resources live